### ISSUE TITLE
Allow the source lines to be of arbitrary length

### ISSUE THREAD

**Issue by arjenmarkus** [j3-fortran/fortran_proposals#133]
It is a minor issue, but sometimes the restriction of at most 132 characters hinders the programmer - or some program generating source code from specifications (I am specifically thinking here of my "flemon"' parser generator). 
Would it not be nice to not have to worry too much about the line length? Note that I do not intend to write code with lines of more than 100 characters or whatever fits nicely on the screen (or the printout). It is just that in particular situations - generated code or deeply nested code - you run into the right-hand margin.

**Comment by certik** [comment]
I am pretty sure we discussed this at the last October Committee meeting and I thought this was generally approved, but I can't find the proposal right now. (Starting this February meeting, I plan to rigorously track every proposal and the associated discussion at GitHub issues here.)

**Comment by FortranFan** [comment]
> I am pretty sure we discussed this at the last October Committee meeting and I thought this was generally approved, but I can't find the proposal right now. (Starting this February meeting, I plan to rigorously track every proposal and the associated discussion at GitHub issues here.)


See https://j3-fortran.org/doc/year/19/19-138r1.txt.  The summary is

> 3 Specifications
>
> A source line contains at most ten thousand characters.
> 
> A source statement contains at most one million characters.
> 
> Processors are required to be able to report too-long lines.
>
> Continuation is unchanged, except that the limit of 255 is removed.


WG5 approved this for Fortran 202X at the Tokyo meeting August 2019..

**Comment by arjenmarkus** [comment]
Perfect :)

**Comment by FortranFan** [comment]
@arjenmarkus wrote:

> Perfect :)

Can this issue be marked as "Completed" (or some such)?!

**Comment by arjenmarkus** [comment]
As this topic has already been discussed in the WG5 group, it can be closed.

**Comment by arjenmarkus** [comment]
Just done so.

Op vr 10 jan. 2020 om 14:59 schreef FortranFan <notifications@github.com>:

**Comment by milancurcic** [comment]
I recommend keeping this issue open, despite there being an existing proposal and discussion at meeting. The rationale for it is that the issue doesn't get buried and people from the community can more easily discover it, support it, and give their two cents. By closing the issue, we're shutting down this side of the conversation. 

Let's keep any issue open until it officially enters the language standard. If there is an existing proposal and discussion already, we can use a tag to mark this "in progress" or similar. @certik what do you think?

**Comment by certik** [comment]
Yes, let's definitely keep the issue open until it becomes part of the standard. I added the appropriate labels for now.
